[Fix] 401 redirection due to error in API calls

taylorz opened the pull request:
### Summary 

After updating much of the data fetching to useSWR some of the `put` api calls, [most](https://app.highlight.io/5415/sessions/b7S0aCD6QcmMS3XP5fR1XzG4B2J4?network-resource-id=7&page=1&query=and%7C%7Ccustom_processed%2Cis%2Ctrue) [notably](https://app.highlight.io/5415/sessions/m5cpPyxS47FEWp8YhWnUuDFxbyCq?network-resource-id=42&page=1&query=and%7C%7Ccustom_processed%2Cis%2Ctrue) those in `api/steps` would erroneously log the user out when an trying to assign a task. 

This PR moves the definition of `config` directly into each individual API call rather than globally doing so in `base` and also ensures that all `put` calls are passed an empty object for params so that the call doesn't fail.

taylorz committed:
Merge pull request #454 from keithtom/production

[Backmerge] Prod ->Staging

taylorz committed:
Merge pull request #455 from keithtom/staging

[Backmerge] Staging -> Dev

taylorz committed:
trying to pass config in task api calls

taylorz committed:
remove import of unused library to avoid illegal invocation error

taylorz committed:
pass config to task api calls differently

taylorz committed:
add params for put calls in api directory and add clarifying comments

styliii commented on api/schools.js:
```diff
@@ -1,20 +1,30 @@
+    config.params = params;
     return schoolsApi
-      .get(`/${schoolId}`, { params: params })
+      .get(`/${schoolId}`, config)
```
Have you tested this? Just want to make sure the params are being passed in as expected.

taylorz replied:
I do see the params being passed in the call when I navigate to the schoolId page!

styliii commented on pages/ssj/[workflow]/[phase]/[milestone]/index.js:
```diff
@@ -2,14 +2,14 @@ import { useRouter } from "next/router";
 import Skeleton from "@mui/material/Skeleton";
 
-import useAuth from "@lib/utils/useAuth";
+// import useAuth from "@lib/utils/useAuth";
```
Are you sure you want to keep this commented out?

taylorz replied:
THanks for catching, do not want to keep commented

styliii commented:
I think `loginEmailLink` needs a config being passed in.

What do you think of completely getting rid of 
```
  if (token) {
    // console.log("token from cookies", jwt_decode(token).sub)
    config.headers["Authorization"] = token;
  } else if (axios.defaults.headers.common["Authorization"]) {
    // console.log("token set from axios global", jwt_decode(axios.defaults.headers.common['Authorization']).sub);
  }
```
in `base.js`? This would prove to us that all the functions that need a auth header passed in was caught.

styliii commented on api/people.js:
```diff
@@ -32,6 +40,7 @@ async function update(personId, personParams) {
 }
 
 // filter example: {{ops_guide: true, rgl: true}}
+// TODO update with SWR hook
```
Do we need to pass in a config into this function?

taylorz replied:
Yes! Adjusted

taylorz commented:
> I think `loginEmailLink` needs a config being passed in.
> 
> What do you think of completely getting rid of
> 
> ```
>   if (token) {
>     // console.log("token from cookies", jwt_decode(token).sub)
>     config.headers["Authorization"] = token;
>   } else if (axios.defaults.headers.common["Authorization"]) {
>     // console.log("token set from axios global", jwt_decode(axios.defaults.headers.common['Authorization']).sub);
>   }
> ```
> 
> in `base.js`? This would prove to us that all the functions that need a auth header passed in was caught.

Added config to `loginEmailLink` thanks for catching-- and updated the people api call with config, thank you!
Also fully remove the code in question in api/base

taylorz committed:
address helpful comments

taylorz committed:
utilize hooks in onboarding

taylorz committed:
uncomment useAuth hook on ssj dash
